tests/unit/labels/LabelsEncoder_unittest.cc

```diff
@@ -0,0 +1,66 @@
+/*
+ * This software is distributed under BSD 3-clause license (see LICENSE file).
+ *
+ * Authors: Yuhui Liu
+ */
+
+#include <gtest/gtest.h>
+#include <shogun/labels/BinaryLabelEncoder.h>
+#include <shogun/labels/LabelEncoder.h>
+#include <shogun/labels/MulticlassLabels.h>
+#include <shogun/labels/MulticlassLabelsEncoder.h>
+using namespace shogun;
+
+TEST(BinaryLabelEncoder, fit_transform)
+{
+	auto label_encoder = std::make_shared<BinaryLabelEncoder>();
+	SGVector<int32_t> vec{-1, -1, 1, -1, 1};
```

Comment:
what about something with values that are not {-1,1}. Could you also test for something with more than two labels to make sure the exception is thrown?

Reply:
if values are not {-1, 1}, they will be transformed to {-1, 1}, i will add more unite test.

Reply:
I guess you want to test

* binary labels are passed (no-op)
* labels with two unique values are passed (and transformed into binary)
* labels with fewer or more than two unique values are passed (error)